Add docs about token standards (#212)
LiskHQ/lisk-docs: docs/building-on-lisk/token-development/overview.md

---
title: Overview
slug: /token-development
description: 'An introduction into and ERC token standards and token development on Lisk.'
keywords:
  [
    'Lisk',
    'Token development',
    'Deploy token',
    'ERC',
    'EIP',
    'ERC-20',
    'ERC-721',
    'ERC-1155',
    'NFT',
    'Fungible token',
    'Hybrid token',
  ]
---

# Token development
This page is intended for token issuers who wish to create a new ERC-20 contract on Lisk.

> It includes an explanation of ERCs, a summary of the most important token standards and examples how to deploy these tokens on Lisk.

Reviewer: ```suggestion
It includes an explanation of ERCs, a summary of the most important token standards and examples of how to deploy these tokens on Lisk.
```

> It includes an explanation of ERCs, a summary of the most important token standards and examples of how to deploy these tokens on Lisk.

Reviewer 2: ```suggestion
It includes an explanation of ERCs, a summary of the most important token standards, and examples of how to deploy these tokens on Lisk.
```

> In case you already have a token deployed on the Ethereum network, an wish to bridge it to Lisk, please refer to the guide [Bridging an L1 token to Lisk](../add-token-to-lisk/index.md).

Reviewer: ```suggestion
In case you already have a token deployed on the Ethereum network, and wish to bridge it to Lisk, please refer to the guide [Bridging an L1 token to Lisk](../add-token-to-lisk/index.md).
```

## ERC token standards

A standard interface allows any tokens on Ethereum to be re-used by other applications: from wallets to decentralized exchanges.

> **ERCs**(= Ethereum Request for Comments) are a set of application-level standards and conventions, including contract standards such as token standards (ERC-20), name registries (ERC-137), URI schemes, library/package formats, and wallet formats for the Ethereum blockchain.

Reviewer 2: ```suggestion
**ERCs**(Ethereum Request for Comments) are a set of application-level standards and conventions, including contract standards such as token standards (ERC-20), name registries (ERC-137), URI schemes, library/package formats, and wallet formats for the Ethereum blockchain.
```

Following the most popular ERC token standards when creating a new token has several benefits:

- **Increased security:** Let your contract inherit from heavily audited and reviewed implementations of the standard, mitigating the possibility of bugs greatly.

> - **High application compatibility:** Most applications only support the most popular ERC token standards. By following these standars you ensure your token will be compatible with most external application like wallets or decentralized exchanges.

Reviewer: ```suggestion
- **High application compatibility:** Most applications only support the most popular ERC token standards. By following these standards you ensure your token will be compatible with most external application like wallets or decentralized exchanges.
```

> - **High application compatibility:** Most applications only support the most popular ERC token standards. By following these standards you ensure your token will be compatible with most external application like wallets or decentralized exchanges.

Reviewer 2: ```suggestion
- **High application compatibility:** Most applications only support the most popular ERC token standards. By following these standards, you ensure your token will be compatible with most external applications like wallets or decentralized exchanges.
```

> - **Great documentation:** Benefit from the vast amount of tutorials and guides that are available to develop ERC compliant tokens.

Reviewer 2: ```suggestion
- **Great documentation:** Benefit from the vast number of tutorials and guides that are available to develop ERC-compliant tokens.
```

> ERCs are a subcategory of **EIPs**(= Ethereum Inprovemnt Proposals).

Reviewer: ```suggestion
ERCs are a subcategory of **EIPs**(= Ethereum Improvement Proposals).
```

> ERCs are a subcategory of **EIPs**(= Ethereum Improvement Proposals).

Reviewer 2: ```suggestion
ERCs are a subcategory of **EIPs**(Ethereum Improvement Proposals).
```

New EIPs are added following the process outlined in [EIP-1](https://eips.ethereum.org/EIPS/eip-1).

> The full list of [ERC proposals](https://eips.ethereum.org/erc).

Reviewer 2: ```suggestion
Here's the full list of [ERC proposals](https://eips.ethereum.org/erc).
```

> A summary of some interesting ERC tokens standars can be found below.

Reviewer: ```suggestion
A summary of some interesting ERC tokens standards can be found below.
```

> A summary of some interesting ERC tokens standards can be found below.

Reviewer 2: ```suggestion
A summary of some interesting ERC token standards can be found below.
```

Any token is exactly equal to any other token; no tokens have special rights or behavior associated with them.
This makes [ERC-20](https://eips.ethereum.org/EIPS/eip-20) tokens useful for things like a medium of exchange currency, voting rights, staking, and more.

### Guides
[How to deploy a new ERC-20 token on Lisk](deploy-erc-20.md)

### Further reading
- [Understand the ERC-20 token smart contract](https://ethereum.org/en/developers/tutorials/understand-the-erc-20-token-smart-contract/)
- [ERC-20 EIP](https://eips.ethereum.org/EIPS/eip-20)
- [OpenZeppelin: ERC-20 API](https://docs.openzeppelin.com/contracts/3.x/api/token/erc20)
- [OpenZeppelin: ERC-20 contract](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol)
- [Solidity by example: ERC-20](https://solidity-by-example.org/app/erc20/)
- [Ethereum Docs: ERC-20](https://ethereum.org/en/developers/docs/standards/tokens/erc-20/)
- [Alchemy: Complete guide to ERC-20](https://www.alchemy.com/overviews/erc20-solidity)

## ERC-721
[ERC-721](https://eips.ethereum.org/EIPS/eip-721) is a standard for representing ownership of non-fungible tokens.

> Non-fungible tokens(NFTs) are used to represent unique objects like real estate or collectibles, where some items are valued more than others, due to their usefulness, rarity, or other individual characteristics.

Reviewer 2: ```suggestion
Non-fungible tokens(NFTs) are used to represent unique objects like real estate or collectibles, where some items are valued more than others due to their usefulness, rarity, or other individual characteristics.
```

> To represent these unique features on chain, the ERC-721 includes metadata properties that offer information about the token's specific features, such as the title, the creator, and an image preview.

Reviewer 2: ```suggestion
To represent these unique features onchain, the ERC-721 includes metadata properties that offer information about the token's specific features, such as the title, the creator, and an image preview.
```
